[iOS] WirelessPlaybackMediaPlayerEnabled can't be enabled at runtime

aestes opened the pull request:
#### 2db7bcd445af5bfc302b39a462e604b08045ec5d
<pre>
[iOS] WirelessPlaybackMediaPlayerEnabled can&apos;t be enabled at runtime
<a href="https://bugs.webkit.org/show_bug.cgi?id=304360">https://bugs.webkit.org/show_bug.cgi?id=304360</a>
<a href="https://rdar.apple.com/166734272">rdar://166734272</a>

Reviewed by Jean-Yves Avenard.

Fixed a bug where the wrong condition was specified in the definition of
WirelessPlaybackMediaPlayerEnabled. While here:
1. Moved the WebCore runtime setting from DeprecatedGlobalSettings to MediaStrategy.
2. Called MediaStrategy::resetMediaEngines() when WirelessPlaybackMediaPlayerEnabled or
   MockMediaDeviceRouteControllerEnabled changes.
3. Disabled MockMediaDeviceRouteControllerEnabled and cleared MediaDeviceRouteController&apos;s client
   in Internals::resetToConsistentState.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/page/DeprecatedGlobalSettings.h:
(WebCore::DeprecatedGlobalSettings::setWirelessPlaybackMediaPlayerEnabled): Deleted.
(WebCore::DeprecatedGlobalSettings::isWirelessPlaybackMediaPlayerEnabled): Deleted.
* Source/WebCore/platform/MediaStrategy.cpp:
(WebCore::MediaStrategy::setWirelessPlaybackMediaPlayerEnabled):
(WebCore::MediaStrategy::wirelessPlaybackMediaPlayerEnabled const):
* Source/WebCore/platform/MediaStrategy.h:
* Source/WebCore/platform/audio/ios/MediaDeviceRouteController.h:
* Source/WebCore/platform/audio/ios/MediaDeviceRouteController.mm:
(WebCore::setMockMediaDeviceRouteControllerEnabled):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::WTF_REQUIRES_LOCK):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
* Source/WebCore/testing/Internals.idl:
* Source/WebCore/testing/MockMediaDeviceRoute.idl:
* Source/WebCore/testing/MockMediaDeviceRouteController.idl:
* Source/WebCore/testing/MockMediaDeviceRouteController.mm:
(WebCore::MockMediaDeviceRouteController::setEnabled):
(WebCore::MockMediaDeviceRouteController::activateRoute):
(WebCore::MockMediaDeviceRouteController::deactivateRoute):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updatePreferences):

Canonical link: <a href="https://commits.webkit.org/304788@main">https://commits.webkit.org/304788@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca6e2a2e5dbb77c248c4081a2beaa902ad98dd4f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136217 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8573 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47497 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143927 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/89185 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138089 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9248 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8418 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104154 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/71990 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139163 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6725 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122059 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84984 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/19f41a04-72b6-459f-8f62-5006c3ef81c4) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6390 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/4048 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4521 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/128175 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115675 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40260 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146672 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/134702 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8257 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40827 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/112497 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8273 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6935 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112837 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28716 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6314 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118364 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/62244 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8305 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36421 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/167481 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8023 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/71864 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43700 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8244 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8097 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebCore/platform/MediaStrategy.h:
```diff
@@ -78,11 +78,21 @@ class WEBCORE_EXPORT MediaStrategy : public CanMakeThreadSafeCheckedPtr<MediaStr
 
+#if ENABLE(WIRELESS_PLAYBACK_MEDIA_PLAYER)
+    void setWirelessPlaybackMediaPlayerEnabled(bool);
+    bool wirelessPlaybackMediaPlayerEnabled() const;
```
this will enabled the ability to turn it on in WK1 ; maybe `MediaStrategy::wirelessPlaybackMediaPlayerEnabled` should be made virtual, returning false by default and the actual code put in WebKitLLWebMediaStrategy

aestes replied:
As we talked about offline, I don't think this is necessary. UnifiedWebPreferences is the source of truth for this feature flag, and there it will remain off by default in WebKitLegacy (also, only WebKit2 code calls `setWirelessPlaybackMediaPlayerEnabled()`).

jyavenard reviewed (approved):
r=m with comment

aestes committed:
[iOS] WirelessPlaybackMediaPlayerEnabled can't be enabled at runtime
https://bugs.webkit.org/show_bug.cgi?id=304360
rdar://166734272

Reviewed by Jean-Yves Avenard.

Fixed a bug where the wrong condition was specified in the definition of
WirelessPlaybackMediaPlayerEnabled. While here:
1. Moved the WebCore runtime setting from DeprecatedGlobalSettings to MediaStrategy.
2. Called MediaStrategy::resetMediaEngines() when WirelessPlaybackMediaPlayerEnabled or
   MockMediaDeviceRouteControllerEnabled changes.
3. Disabled MockMediaDeviceRouteControllerEnabled and cleared MediaDeviceRouteController's client
   in Internals::resetToConsistentState.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/page/DeprecatedGlobalSettings.h:
(WebCore::DeprecatedGlobalSettings::setWirelessPlaybackMediaPlayerEnabled): Deleted.
(WebCore::DeprecatedGlobalSettings::isWirelessPlaybackMediaPlayerEnabled): Deleted.
* Source/WebCore/platform/MediaStrategy.cpp:
(WebCore::MediaStrategy::setWirelessPlaybackMediaPlayerEnabled):
(WebCore::MediaStrategy::wirelessPlaybackMediaPlayerEnabled const):
* Source/WebCore/platform/MediaStrategy.h:
* Source/WebCore/platform/audio/ios/MediaDeviceRouteController.h:
* Source/WebCore/platform/audio/ios/MediaDeviceRouteController.mm:
(WebCore::setMockMediaDeviceRouteControllerEnabled):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::WTF_REQUIRES_LOCK):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
* Source/WebCore/testing/Internals.idl:
* Source/WebCore/testing/MockMediaDeviceRoute.idl:
* Source/WebCore/testing/MockMediaDeviceRouteController.idl:
* Source/WebCore/testing/MockMediaDeviceRouteController.mm:
(WebCore::MockMediaDeviceRouteController::setEnabled):
(WebCore::MockMediaDeviceRouteController::activateRoute):
(WebCore::MockMediaDeviceRouteController::deactivateRoute):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updatePreferences):

Canonical link: https://commits.webkit.org/304788@main
